src/nccl_ofi_rdma.c

```diff
@@ -2528,7 +2528,7 @@ static inline int reg_mr_on_device(nccl_net_ofi_rdma_ep_t *ep,
 		goto exit;
 	}
 
-	ret = register_rail_mr_buffer(get_device_rail(device, 0)->domain, ep->control_rail.ofi_ep,
+	ret = register_rail_mr_buffer(ep->rails[0].domain, ep->control_rail.ofi_ep,
```

Comment:
I think `ep->control_rail.domain` would be the morally correct thing to use here, even though today they will be identical.

Reply:
Fixed, tested, and merged.